### DISCUSSION TITLE
Posting an empty array in a Form errors with 422

### DISCUSSION THREAD

**Discussion by jedahan** [fastapi/fastapi#6065] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# python3 -m pip install fastapi pytest
# python3 -m pytest tests/post.py

from typing import Optional

from fastapi import Form, FastAPI, Query
from fastapi.testclient import TestClient

app = FastAPI()

@app.post("/list")
def read_query(q: list[Optional[str]] = Form()):
    return q

client = TestClient(app)

def test_full_list_valid():
    response = client.post("/list", data={"q": ["hi", "there"]})
    assert response.status_code == 200
    assert response.json() == ["hi", "there"]

def test_empty_list_valid():
    response = client.post("/list", data={"q": []})
    assert response.status_code == 200
    assert response.json() == []
```

### Description

I would expect all tests to pass

### Operating System

macOS

### FastAPI Version

0.85.2

### Python Version

Python 3.10.9

**Comment by jedahan** [comment]
This only somewhat overlaps with https://github.com/tiangolo/fastapi/issues/3506

**Comment by YuriiMotov** [comment]
`response = client.post("/list", data={"q": []})` actually sends empty body

```
    response = client.post("/list", data={"q": []})
    print(response.request.content)
    
    # b''
```

If you want to allow passing empty body, you need to specify default value
